plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngineIndexer.ts

```diff
@@ -95,7 +96,7 @@ export class ElasticSearchSearchEngineIndexer extends BatchSearchEngineIndexer {
           index: { _index: that.indexName },
         };
       },
-      refreshOnCompletion: that.indexName,
+      refreshOnCompletion: !options.skipRefresh && that.indexName,
```

Comment:
if options.skipRefresh is `true` here, `refreshOnCompletion` will be set to `false`. Are we sure that's alright? Or do we want to handle this differently? Maybe we don't want to pass `refreshOnCompletion` instead?

Reply:
Also, does this really need to be on the indexer at all? Would it be possible to leave the indexer completely unchanged and instead leverage the fact that the client is wrapped up and segregates opensearch already?

https://github.com/backstage/backstage/blob/b8ce15b1e5bfa2f5184dc61c96af6c45addcd1aa/plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchClientWrapper.ts#L122

Could it just exclude the refreshOnCompletion field in one of the paths there selectively?

Reply:
> if `options.skipRefresh` is `true` here, `refreshOnCompletion` will be set to `false`. Are we sure that's alright? Or do we want to handle this differently? Maybe we don't want to pass `refreshOnCompletion` instead?

It is an optional field but the default is `false` so net effect is the same... https://opensearch.org/docs/latest/api-reference/document-apis/bulk/

Reply:
> Also, does this really need to be on the indexer at all? Would it be possible to leave the indexer completely unchanged and instead leverage the fact that the client is wrapped up and segregates opensearch already?
> 
> https://github.com/backstage/backstage/blob/b8ce15b1e5bfa2f5184dc61c96af6c45addcd1aa/plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchClientWrapper.ts#L122
> 
> Could it just exclude the refreshOnCompletion field in one of the paths there selectively?

I don't think so.  OpenSearch supports refresh.  However, AWS OpenSearch Serverless does not.

Reply:
Ready for another round of review.